intro_to_R_tidyverse/intro_to_tidyverse.Rmd

```diff
@@ -0,0 +1,185 @@
+But if you needed to install it or other packages available on CRAN, you 
+do it using the `install.packages` function like this: `install.packages("tidyverse")`
+```{r Load tidyverse}
+library(tidyverse)
+```
+
+"metadata.tsv" is a file we will use for practicing the tidyverse.
+This metadata is from GSE30074 [refine.bio](https://www.refine.bio), a gene 
+expression dataset from medulloblastoma.
+  
+First let's read in the dataset using the tidyverse package: `readr`.
+Although base R has read-in functions, the `readr` functions are faster and more
+straight forward to use so we are going to use those here. 
+Because the file we are reading in is a .tsv file we will be using the 
+`read_tsv` function. 
+But note that there are analogous functions for .csv files (`read_csv`) and 
+other files types.
+
+### About directories
+Before we can import the file we need though, we should double check what
+directory R is in, aka what our current *directory* is. 
+We can do this by using the `getwd()` function, which will tell us what folder
+we are in. 
+Then we can use the `dir()` function to have R list the files in our current 
+directory. 
+What we should see is the file `metadata.tsv` show up as one of the files when 
+we run `dir()`.
+If for some reason we were not in the correct directory, we can either set our
+directory to somewhere else using `setwd()` OR we can tell R to look for the 
+file `metadata.tsv` somewhere else using a relative `file.path()`.
+We will discuss and practice these concepts more in other modules as we go along.
+```{r}
+# Let's check what directory we are in:
+getwd()
+
+# Let's check what files are here
+dir()
+
+# The file we are reading in is our argument 
+metadata <- read_tsv("metadata.tsv")
+```
+
+## dplyr pipes
+A nifty thing with the tidyverse is the pipes: `%>%`
+This handy thing allows you to funnel the result of one expression to the next
+making your code a little more streamlined.
+
+For example, the output from this:  
+```
+filter(metadata, sex == "F")
+```  
+...is the same as the output from this:  
+```
+metadata %>% filter(sex == "F")
+```  
+  
+This makes it so your code is cleaner and easier to read. 
+Let's look at an example with our metadata of of how the same 
+functions look with or without pipes:
+
+*Example 1:* without pipes:   
+```{r}
+metadata_edit <- metadata
+metadata_edit <- arrange(metadata_edit, histology)
+metadata_edit <- filter(metadata_edit, follow_up_mths > 50)
+metadata_edit <- select(metadata_edit, sample_id, sex, organism)
+```
+UGH, we have to repeat and assign `special.mtcars` so many times here! 
+It's annoying and makes it harder for people to read. 
+
+*Example 2:* Same result as 1 but with pipes!
+```{r}
+# Example of the same modifications as above but with pipes!
+metadata_edit <- metadata %>%
+                 arrange(histology) %>%
+                 filter(follow_up_mths > 50) %>%
+                 select(sample_id, sex, organism)
```

Comment:
Add a block where we're viewing `metadata_edit`?

Reply:
Perhaps in both spots (here and at ~line 95). Might be nice to do something with `all.equal` as well.

Reply:
Cool. Will do.